02-frequency_filter.py

```diff
@@ -92,8 +92,7 @@ def run_filter(subject, run=None, session=None):
         raw.plot(n_channels=50, butterfly=True)
 
         # plot power spectral densitiy
-        raw.plot_psd(area_mode='range', tmin=10.0, tmax=100.0,
-                     fmin=0., fmax=50., average=True)
+        raw.plot_psd(average=True, area_mode='range', n_fft=len(raw.times))
```

Comment:
this is not the right fix. You DO NOT want to compute the FFT of the full signal but rather average more windows in the Welch method so do not specify tmin and tmax

I would do:

`raw.plot_psd(area_mode='range', fmin=0., fmax=50., average=False)`

Reply:
Ah you see, I was struggling with the meaning of those parameters :) So what I actually want to change are `nperseg` and `noverlap` passed to `scipy.signal.welch()`, right?
https://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.signal.welch.html

Reply:
yes but these parameters are not exposed as is in plot_psd